src/context/auth-context.js

```diff
@@ -0,0 +1,31 @@
+import { createContext, useState, useContext, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+const AuthContext = createContext();
+const useAuth = () => useContext(AuthContext);
+
+const AuthProvider = ({ children }) => {
+  const navigate = useNavigate();
+  const [loggedIn, setLoggedIn] = useState(false);
+  const userToken = localStorage.getItem("userToken");
+
+  useEffect(() => {
+    if (userToken) {
+      setLoggedIn(true);
+    }
+  }, [userToken]);
+
+  const logoutHandler = () => {
```

Comment:
suggestion: you can write this handler in the page where you are created logout button.

Reply:
Done